src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -280,81 +350,65 @@ private static boolean isInLambda(DetailAST paramDef) {
      */
     private static DetailAST findClassOrConstructorOrMethodInWhichItIsDefined(DetailAST ast) {
         DetailAST astTraverse = ast;
-        while (!(astTraverse.getType() == TokenTypes.METHOD_DEF
-                || astTraverse.getType() == TokenTypes.CLASS_DEF
-                || astTraverse.getType() == TokenTypes.ENUM_DEF
-                || astTraverse.getType() == TokenTypes.CTOR_DEF)) {
+        while (astTraverse.getType() != TokenTypes.METHOD_DEF
+                && astTraverse.getType() != TokenTypes.CLASS_DEF
+                && astTraverse.getType() != TokenTypes.ENUM_DEF
+                && astTraverse.getType() != TokenTypes.CTOR_DEF) {
             astTraverse = astTraverse.getParent();
         }
         return astTraverse;
     }
 
     /**
-     * Check if both the Variable are same.
+     * Check if both the Variables are same.
      * @param ast1 Variable to compare
      * @param ast2 Variable to compare
-     * @return true if both the variable are same, otherwise false
+     * @return true if both the variables are same, otherwise false
      */
     private static boolean isSameVariables(DetailAST ast1, DetailAST ast2) {
         final DetailAST classOrMethodOfAst1 =
             findClassOrConstructorOrMethodInWhichItIsDefined(ast1);
         final DetailAST classOrMethodOfAst2 =
             findClassOrConstructorOrMethodInWhichItIsDefined(ast2);
-
-        final String identifierOfAst1 =
-            classOrMethodOfAst1.findFirstToken(TokenTypes.IDENT).getText();
-        final String identifierOfAst2 =
-            classOrMethodOfAst2.findFirstToken(TokenTypes.IDENT).getText();
-
-        return identifierOfAst1.equals(identifierOfAst2);
+        return classOrMethodOfAst1 == classOrMethodOfAst2;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/checkstyle/checkstyle/file/1111847767/issues/source?bid=2193633&fileBranchId=2892126#l373)